### PR TITLE
Add dns_name, hosted_zone_id attributes to Global Accelerator for Route53 ALIASes

### LINKED ISSUE #11579: Provide DNS name from Global Accelerator resource
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Route53 supports ALIAS records pointing at Global Accelerator (GA) listeners _but_ the ALIAS needs to point to the DNS name of the GA. The current Terraform resource doesn't export a DNS name/zone attribute, so you can't wire up an alias with Terraform.

(The workaround is to create an A record pointing at the static `ip_addresses` of the GA, but ALIASes are a nice bit of self-documentation instead of having magic addresses that you have to cross-reference.)

### New or Affected Resource(s)

* `aws_globalaccelerator_accelerator`

### Potential Terraform Configuration

```hcl
resource "aws_globalaccelerator_accelerator" "example" {
  name            = "example"
  ip_address_type = "IPV4"
  enabled         = true
}

resource "aws_route53_record" "record" {
  zone_id = "SOME_ZONE_ID"
  name    = "ga.example.com"

  alias {
    name                   = aws_globalaccelerator_accelerator.example.dns_name
    zone_id                = aws_globalaccelerator_accelerator.example.zone_id
    evaluate_target_health = false
  }
}
```

### References

* https://docs.aws.amazon.com/global-accelerator/latest/api/API_CreateAccelerator.html (returns `DNSName`). 
* https://docs.aws.amazon.com/global-accelerator/latest/dg/about-accelerators.dns-addressing.html
* It's not clear to me where `zone_id` would come from

### PR DESCRIPTION
**Pull request by wlonkly**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #11579

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
ENHANCEMENTS
* resource/aws_globalaccelerator_accelerator: Add `dns_name` and `hosted_zone_id` attributes [GH-11579]
```

Output from acceptance testing:

```
$ make testacc TEST=./aws TESTARGS='-run=GlobalAcceleratorAccelerator'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=GlobalAcceleratorAccelerator -timeout 120m
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_basic
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_basic
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_update
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_update
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_attributes
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_attributes
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_basic
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_update
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_attributes
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_basic (82.84s)
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_attributes (86.71s)
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_update (116.89s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	117.005
```

### CODE REVIEW

File: aws/resource_aws_globalaccelerator_accelerator.go [hunk 1]
```diff
@@ -13,6 +13,10 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
 )
 
+// Global Route53 Zone ID for Global Accelerators, exported as a
+// convenience attribute for Route53 aliases.
+const globalAcceleratorRoute53ZoneID = "Z2BJ6XQ5FK7U4H"
```
Comment (wlonkly) [inline aws/resource_aws_globalaccelerator_accelerator.go]: Obtained from http://docs.aws.amazon.com/Route53/latest/APIReference/CreateAliasRRSAPI.html, followed the pattern used by `aws_cloudfront_distribution` which also has a global Route53 zone id.

Comment (voidlily) [reply aws/resource_aws_globalaccelerator_accelerator.go]: That link didn't work for me, but the zone id is also mentioned on https://docs.aws.amazon.com/Route53/latest/APIReference/API_AliasTarget.html

Comment (wlonkly) [reply aws/resource_aws_globalaccelerator_accelerator.go]: That's the one, thanks!

**Comment by thinders-accenture** [comment]
We could use the attributes this PR would add. Is there anything preventing this PR from being merged?

**Comment by anthonyelliott-pillar** [comment]
@bflad does this PR look good to you? It sure would help us out!

File: aws/resource_aws_globalaccelerator_accelerator.go [hunk 2]
```diff
@@ -13,6 +13,10 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
 )
 
+// Global Route53 Zone ID for Global Accelerators, exported as a
+// convenience attribute for Route53 aliases.
```
Comment (paultyng) [inline aws/resource_aws_globalaccelerator_accelerator.go]:
```suggestion
// convenience attribute for Route53 aliases (see 
// https://docs.aws.amazon.com/Route53/latest/APIReference/API_AliasTarget.html).
```

File: website/docs/r/globalaccelerator_accelerator.markdown [hunk 3]
```diff
@@ -46,13 +46,19 @@ The following arguments are supported:
...
+  is simply an alias for the zone ID `Z2BJ6XQ5FK7U4H`.
 * `ip_sets` - IP address set associated with the accelerator.
 
 **ip_sets** exports the following attributes:
 
-* `ip_addresses` - The array of IP addresses in the IP address set.
+* `ip_addresses` - A list of IP addresses in the IP address set.
 * `ip_family` - The types of IP addresses included in this IP set.
 
+[1]: http://docs.aws.amazon.com/Route53/latest/APIReference/CreateAliasRRSAPI.html
```
Comment (bflad) [inline website/docs/r/globalaccelerator_accelerator.markdown]:
```suggestion
[1]: https://docs.aws.amazon.com/Route53/latest/APIReference/API_AliasTarget.html
```

**Pull request merged**